docs/architecture.md

```diff
@@ -1,188 +1,264 @@
-the Kubewarden stack:
+Kubernetes discovers Kubewarden's Webhook endpoints using `kubewarden-controller`.
+This works by Kubewarden registering either a `MutatingWebhookConfiguration`
+or a `ValidatingWebhookConfiguration` object with Kubernetes.
+
+<!--TODO:@vicuad points out we need a paragraph about the Audit Scanner.-->
+
+The diagram shows the architecture of a cluster running the Kubewarden stack:
+
+<!--TODO:To be clear. The stack refers to the entire diagram rather that the
+stacked components for e.g., CAP.-->
+
+<!--TODO:Do we really need the full diagram at this point. Maybe start
+with the simpler one in the next section-->
 
 ![Full architecture](/img/architecture.png)
 
-## Journey of a Kubewarden policy
+## The journey of a Kubewarden policy
+
+The architecture diagram appears complex at first.
+This section covers it step by step.
 
-The architecture diagram from above can be intimidating at first, this
-section explains it step by step.
+### Default policy server
 
-### Default Policy Server
+On a new cluster, the Kubewarden components defined are:
 
-On a fresh new cluster, the Kubewarden components defined are its Custom
-Resource Definitions, the `kubewarden-controller` Deployment and a `PolicyServer`
-Custom Resource named `default`.
+<!--TODO:How are CRDs shown in the diagram-->
+
+- the Custom Resource Definitions (CRD)
+- the `kubewarden-controller` Deployment
+- a `PolicyServer` Custom Resource named `default`.
 
 ![Defining the first ClusterAdmissionPolicy resource](/img/architecture_sequence_01.png)
 
-`kubewarden-controller` notices the default `PolicyServer` resource and, as a result of that,
+<!--TODO:For the next paragraph. What type of architecture diagram is this? In the
+diagram, which is `policy-server`. The aquamarine box with 'Kubewarden Policy
+Server'? Is 'Deployment' capitalization correct? What does 'notices' mean
+exactly? For the diagram, what's the difference between solid box, dashed box,
+white box, aquamarine box? What are the arrows showing, should they be
+labelled?-->
+
+When the `kubewarden-controller` notices the default `PolicyServer` resource,
 it creates a Deployment of the `policy-server` component.
 
-As stated above, Kubewarden works as a Kubernetes Admission Webhook. Kubernetes
-dictates that all the Webhook endpoints must be secured with TLS.
-`kubewarden-controller` takes care of setting up this secure communication
-by doing these steps:
+Kubewarden works as a Kubernetes Admission Webhook.
+Kubernetes specifies using TLS to secure all Webhook endpoints.
+The `kubewarden-controller` sets up this secure communication
+by:
+
+1. Generating a self-signed Certificate Authority
+1. Use this CA to generate a TLS certificate key for the `policy-server` Service.
 
-1. Generate a self-signed Certificate Authority
-1. Use this CA to generate a TLS certificate and a TLS key for the
-   `policy-server` Service.
+<!--The capitalization on Secret is ok? Maybe in ``?-->
```

Comment:
it is, as it is a Kubernetes resource (see [k8s api docs](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#secret-v1-core)).

Reply:
Thanks